OTAnalytics/plugin_parser/streaming_parser.py

```diff
@@ -0,0 +1,349 @@
+import bz2
+from abc import ABC, abstractmethod
+from itertools import islice
+from pathlib import Path
+from typing import Any, Callable, Iterable, Iterator
+
+import ijson
+
+from OTAnalytics.application.datastore import DetectionMetadata, VideoMetadata
+from OTAnalytics.application.state import TracksMetadata, VideosMetadata
+from OTAnalytics.domain.progress import ProgressbarBuilder
+from OTAnalytics.domain.track import (
+    Detection,
+    Track,
+    TrackClassificationCalculator,
+    TrackId,
+)
+from OTAnalytics.domain.track_dataset import TrackDataset
+from OTAnalytics.plugin_datastore.track_geometry_store.pygeos_store import (
+    PygeosTrackGeometryDataset,
+)
+from OTAnalytics.plugin_datastore.track_store import (
+    PandasByMaxConfidence,
+    PandasTrackDataset,
+)
+from OTAnalytics.plugin_parser import ottrk_dataformat as ottrk_format
+from OTAnalytics.plugin_parser.json_parser import parse_json_bz2
+from OTAnalytics.plugin_parser.otvision_parser import (
+    OttrkFormatFixer,
+    OttrkParser,
+    TrackIdGenerator,
+    TrackLengthLimit,
+    create_python_track,
+    parse_python_detection,
+)
+from OTAnalytics.plugin_progress.tqdm_progressbar import TqdmBuilder
+
+RawDetectionData = list[dict]
+RawVideoMetadata = dict
+RawFileData = tuple[RawDetectionData, RawVideoMetadata, TrackIdGenerator]
+
+
+def parse_json_bz2_events(path: Path) -> Iterable[tuple[str, str, str]]:
+    """
+    Provide lazy data stream reading the bzip2 compressed file
+    at the given path and interpreting it as json objects.
+    """
+    stream = bz2.BZ2File(path)
```

Comment:
Do we need to use the with context manager to ensure the file handle is properly closed at the end of reading?